iphone/Maps/UI/Help/AboutController/AboutController.swift

```diff
@@ -0,0 +1,535 @@
+
+    NSLayoutConstraint.activate([
+      scrollView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+      scrollView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+      scrollView.topAnchor.constraint(equalTo: view.topAnchor),
+      scrollView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+      scrollView.contentLayoutGuide.widthAnchor.constraint(equalTo: scrollView.frameLayoutGuide.widthAnchor),
+
+      stackView.leadingAnchor.constraint(equalTo: scrollView.contentLayoutGuide.leadingAnchor, constant: 20),
+      stackView.trailingAnchor.constraint(equalTo: scrollView.contentLayoutGuide.trailingAnchor, constant: -20),
+      stackView.topAnchor.constraint(equalTo: scrollView.contentLayoutGuide.topAnchor),
+      stackView.bottomAnchor.constraint(equalTo: scrollView.contentLayoutGuide.bottomAnchor),
+
+      logoImageView.heightAnchor.constraint(equalToConstant: 64),
+      logoImageView.widthAnchor.constraint(equalTo: logoImageView.heightAnchor),
+
+      additionalInfoStackView.widthAnchor.constraint(equalTo: stackView.widthAnchor),
+
+      osmView.widthAnchor.constraint(equalTo: stackView.widthAnchor),
+
+      infoTableView.widthAnchor.constraint(equalTo: stackView.widthAnchor),
+      infoTableView.heightAnchor.constraint(equalToConstant: Constants.infoTableViewCellHeight * CGFloat(infoTableViewData.count)),
+
+      socialMediaHeaderLabel.leadingAnchor.constraint(equalTo: socialMediaCollectionView.leadingAnchor),
+
+      socialMediaCollectionView.widthAnchor.constraint(equalTo: stackView.widthAnchor),
+      socialMediaCollectionView.contentLayoutGuide.widthAnchor.constraint(equalTo: stackView.widthAnchor),
+      socialMediaCollectionViewHeighConstraint,
+
+      termsOfUseAndPrivacyPolicyView.widthAnchor.constraint(equalTo: stackView.widthAnchor),
+    ])
+    donationView.widthAnchor.constraint(equalTo: stackView.widthAnchor).isActive = isDonateEnabled()
+
+    view.layoutIfNeeded()
+    updateCollection()
+  }
+
+  func updateCollection() {
+    socialMediaCollectionView.collectionViewLayout.invalidateLayout()
+    DispatchQueue.main.async {
+      self.socialMediaCollectionViewHeighConstraint.constant = self.socialMediaCollectionView.collectionViewLayout.collectionViewContentSize.height
+    }
+  }
+
+  func isDonateEnabled() -> Bool {
+    return Settings.donateUrl() != nil
+  }
+
+  func buildInfoTableViewData() -> [AboutInfoTableViewCellModel] {
+    let infoContent: [AboutInfo] = [.faq, .reportMapDataProblem, .reportABug, .news, .volunteer, .rateTheApp]
+    let data = infoContent.map { [weak self] aboutInfo in
+      return AboutInfoTableViewCellModel(title: aboutInfo.title, image: aboutInfo.image, didTapHandler: {
+        switch aboutInfo {
+        case .faq:
+          self?.navigationController?.pushViewController(FaqController(), animated: true)
+        case .reportABug:
+          guard let link = aboutInfo.link else { return }
```

Comment:
This is only for the optional unwrapping because `AboutInfo`'s prop `link: String?`. Some cases have link, and some don't.
Maybe fatalError will be better here?

Reply:
If issues can be discovered and fixed during the development and testing, then it's better to crash in the unexpected case, of course.